k8s/client.go

```diff
@@ -16,27 +25,79 @@ type Client interface {
 	WatchIngresses(Watcher) error
 }
 
-type noopClient struct {
+type impl struct {
```

Comment:
Yep, lots of golang libs I've looked at do that. It's a very common idiom in C++ too.